### ISSUE TITLE
Strings interpreted differently within :dstr

### ISSUE THREAD

**Issue by nex3** [seattlerb/ruby2ruby#32]
If I pass a string containing a backslash to Ruby2Ruby in `:str`, the emitted code produces the same Ruby string when evaluated. For example:

``` ruby
Ruby2Ruby.new.process(s(:str, "\\t")) #=> "\"\\\\t\""
eval(Ruby2Ruby.new.process(s(:str, "\\t"))) == "\\t" #=> true
```

However, if I do the same thing within `:dstr`, the emitted code produces a Ruby string with the backslash escapes resolved:

``` ruby
Ruby2Ruby.new.process(s(:dstr, "\\t")) #=> "\"\\t\""
Ruby2Ruby.new.process(s(:dstr, "", s(:str, "\\t"))) #=> "\"\\t\""
eval(Ruby2Ruby.new.process(s(:dstr, "\\t"))) == "\\t" #=> false
```

Not only is this inconsistent, it breaks roundtripping with RubyParser:

``` ruby
Ruby2Ruby.new.process(RubyParser.new.parse('"\\\\t#{}"')) #=> "\"\\t\#{}\""
Ruby2Ruby.new.process(RubyParser.new.parse('"\\\\t#{}"')) == '"\\\\t#{}"' #=> false
eval(Ruby2Ruby.new.process(RubyParser.new.parse('"\\\\t#{}"'))) == eval('"\\\\t#{}"') #=> false
```

**Comment by zenspider** [comment]
Also... horrible. This is SUCH a doozy.

**Comment by zenspider** [comment]
OK. I've taken a whack at this a couple of times now... I finally got some clarity on the situation. The short explanation was that `process_str` and `process_dstr` were treating their string content differently _sometimes_. I've got a test in place that enumerates all your edge cases described and all the ones that build up to them and does a good job of breaking it down step by step so I could finally see where the problem actually fell.

Long story short. I think I fixed it.

@presidentbeef if you rely on ruby2ruby, you want to put this through its paces

**Comment by presidentbeef** [comment]
Sure, I'll test it out.

**Comment by presidentbeef** [comment]
This doesn't cause any issues with Brakeman.
